LDMP/main_widget.py

```diff
@@ -121,56 +137,210 @@ def setupDatasets(self):
-        alg5G1.run_mode = AlgorithmRunMode.Both
-        alg5G1_details = AlgorithmDetails(name=None, name_details=None, description='This is the main dindicator that bla bla bla bla bla', parent=alg5G1)
-        alg5G1.setDetails(alg5G1_details)
-
-        first.algorithms.append(alg1G1)
-        first.algorithms.append(alg2G1)
-        first.algorithms.append(alg3G1)
-        first.algorithms.append(alg4G1)
-        first.algorithms.append(alg5G1)
-
-        alg1G2 = AlgorithmDescriptor(name='alg1G2', name_details='alg1G2 details', brief_description=None, details=None, parent=second)
-        alg1G2.run_mode = AlgorithmRunMode.Both
-        alg1G2_details = AlgorithmDetails(name=None, name_details=None, description='This is the second dindicator that bla bla bla bla bla', parent=alg1G2)
-        alg1G2.setDetails(alg1G2_details)
-        second.algorithms.append(alg1G2)
-
-        alg1G3 = AlgorithmDescriptor(name='alg1G3', name_details='alg1G3 details', brief_description=None, details=None, parent=third)
-        alg1G3.run_mode = AlgorithmRunMode.NotApplicable
-        alg1G3_details = AlgorithmDetails(name=None, name_details=None, description='This is description for miscellaneous stuffs', parent=alg1G3)
-        alg1G3.setDetails(alg1G3_details)
-        third.algorithms.append(alg1G3)
-
-
-        tree.algorithms.append(first)
-        tree.algorithms.append(second)
-        tree.algorithms.append(third)
+        land_degradation_group = AlgorithmGroup(name=tr('SDG 15.3.1'), name_details=tr('Land degradation'), parent=tree, algorithms=[])
+        urban_change_group = AlgorithmGroup(name=tr('SDG 11.3.1'), name_details=tr('Urban change and land consumption'), parent=tree, algorithms=[])
+        experimental = AlgorithmGroup(name=tr('Experimental'), name_details=None, parent=tree, algorithms=[])
+        total_carbon_group = AlgorithmGroup(name=tr('Total carbon'), name_details=tr('above and belowground, emissions and deforestation'), parent=experimental, algorithms=[])
+        potential_change_group = AlgorithmGroup(name=tr('Potential change in biomass due to restoration'), name_details=tr('above and belowground woody'), parent=experimental, algorithms=[])
+
+        # land_degradation_group
+        land_productivity_alg = AlgorithmDescriptor(
+            name=tr('Land productivity'), 
+            name_details=None,
+            brief_description=None,
+            details=None,
+            parent=land_degradation_group,
+            run_callbacks={AlgorithmRunMode.Remotely: self.dlg_calculate_LD.btn_prod_clicked})
+        land_productivity_alg.run_mode = AlgorithmRunMode.Remotely
+        land_productivity_alg_details = AlgorithmDetails(
+            name=None,
+            name_details=None,
+            description=tr('TODO: Land productivity long description'),
+            parent=land_productivity_alg)
+        land_productivity_alg.setDetails(land_productivity_alg_details)
```

Comment:
This API seems contrived and we ought to make it simpler. 

Do we really need `AlgorithmDetails` to be a class? And what is the meaning of it having `name`, `name_details`,  `parent`? Seems like this complexity is just for satisfying Qt's MVC stuff. IMO we can use something simpler to understand as an algorithm's configuration and then transform it to whatever Qt needs in order to show it in the tree.

Reply:
1: Algorithm details can be more that a simple description... I would leave it as container
2: the original idea was to use details inside a widget (was the first really hugly GUI imlementation)... then simplified the MVC leaved organizing to the TreeView but required to add a class/model level.
The fact that remianed name and name_details is because the original base class had name and name_detaisl becase wasn0't expected to have details ans node... we can remove. But probably we are spending too much time in representation than running the algorithm.Imho a second core part woudl be to refactor and simplify this part... creating a good widget as node (more or less as in Dataasets) but time is short to focus completly in the estetics.